src/convertToObject.js

```diff
@@ -12,7 +12,20 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const styles = sourceString.split(';')
+    .map(item => item.split(':'))
+    .filter(style => style.length > 1)
+    .map(([property, value]) => {
+      return [property.replace('\n', '').trim(), value.trim()];
+    });
+
+  const result = {};
+
+  styles.forEach(([property, value]) => {
+    result[property] = value;
+  });
```

Comment:
```suggestion
    .map(([property, value]) => {
      return [property.replace('\n', '').trim(), value.trim()];
    });

  const result = {};

  styles.forEach(([property, value]) => {
    result[property] = value;
  });
```
Мне кажется последний перебор map - лишний. Убрать перенос строки и пробелы можно во время записи свойств в объект.

Reply:
agree with @Andrey203